File: src/main/java/com/gmail/borlandlp/minigamesdtools/gui/hotbar/items/inventory_gui/InventoryPageOpenRequester.java
package com.gmail.borlandlp.minigamesdtools.gui.hotbar.items.inventory_gui;

import com.gmail.borlandlp.minigamesdtools.events.PlayerRequestOpenInvGUIEvent;
import com.gmail.borlandlp.minigamesdtools.gui.hotbar.items.SlotItem;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;

public class InventoryPageOpenRequester extends SlotItem {
    private String pageId;

    public String getPageId() {
        return pageId;
    }

    public void setPageId(String pageId) {
        this.pageId = pageId;
    }

    @Override
    public boolean use(Player player) {
        PlayerRequestOpenInvGUIEvent event = new PlayerRequestOpenInvGUIEvent(player, this.getPageId());
        Bukkit.getPluginManager().callEvent(event);
        return true;
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/gui/hotbar/items/inventory_gui/InventoryPageOpenRequester.kt
package com.gmail.borlandlp.minigamesdtools.gui.hotbar.items.inventory_gui

import com.gmail.borlandlp.minigamesdtools.events.PlayerRequestOpenInvGUIEvent
import com.gmail.borlandlp.minigamesdtools.gui.hotbar.items.SlotItem
import org.bukkit.Bukkit
import org.bukkit.entity.Player

class InventoryPageOpenRequester : SlotItem() {
    var pageId: String? = null

    override fun use(player: Player): Boolean {
        val event =
            PlayerRequestOpenInvGUIEvent(player, pageId!!)
        Bukkit.getPluginManager().callEvent(event)
        return true
    }
}